Feature/refactor

Priyanshu-gkm opened the pull request:
- renamed variables
- segregated models, serializers, views and helpers
- changed routes and endpoints
- renamed files and folders

Priyanshu-gkm committed:
refactor: restructured and refactored code

Priyanshu-gkm committed:
refactor: removed user_model.py file(redundant)

AditSoni commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
-.pytest_cache
\ No newline at end of file
+.pytest_cache
+migrations
```
why are migrations in git ignore?

Priyanshu-gkm replied:
removed migrations from gitignore and added the migrations in the latest commit

Priyanshu-gkm committed:
added migrations

Priyanshu-gkm committed:
changed application name to snake case

Priyanshu-gkm committed:
refactor: corrected some variables

Priyanshu-gkm commented:
- changed the response object body to get a standard response
- changed some variables
- changed endpoint DELETE /follow-requests/<id> to DELETE /connections/<id>
- changed some response codes
- removed relative imports with absolute imports

NOTE : test cases were made in a hurry, they are not made according to the standards. Some functions are tested within single test case due to the sequence of events.

Priyanshu-gkm committed:
refactor: modification in response object

Priyanshu-gkm committed:
refactor: user tests,views and permissions

Priyanshu-gkm commented:
- modified permissions 
- refactored user deletion view
- added a new test case file for user authentication - register, login and logout 
- modified test_user.py to contain only test cases related to user operations.

Priyanshu-gkm committed:
refactor: modified test_post_type.py

Priyanshu-gkm commented:
- modified test cases for post_type

Priyanshu-gkm committed:
refactor: modified post tests and view

Priyanshu-gkm commented:
- modified post view , only authenticated users can view the posts
- added more test cases for the /posts endpoint

Priyanshu-gkm committed:
refactor: modified search test cases

Priyanshu-gkm committed:
refactor: modify conections

Priyanshu-gkm commented:
- added new test file for testing connections
- modified response object for responding to follow request

Priyanshu-gkm committed:
refactor: modified feed view and test

Priyanshu-gkm commented:
- created a new file for testing /feed endpoint
- modified feed view

Priyanshu-gkm committed:
refactor: added tests for notification

Priyanshu-gkm commented:
- modified view of notification for redundant code
- added a new test file for testing notifications
- renamed TestConnection -> TestFeed in test_feed.py ( missed in previous commit )

Priyanshu-gkm committed:
refactor: follow request

Priyanshu-gkm commented:
- modified follow request model init function 
- modified follow request view, removed redundant data
- modified follow request test cases, proper segmentation and assertions added

Priyanshu-gkm committed:
added readme

Priyanshu-gkm committed:
renamed readme

Priyanshu-gkm committed:
refactor: username unique
